core/src/test/java/com/microsoft/applicationinsights/telemetry/SanitizerTest.java

```diff
@@ -21,150 +21,145 @@
     private final static String VALID_URL = "http://wwww.microsoft.com/";
     private final static String NON_VALID_URL = "http:sd{@~fsd.s.d.f;fffff";
-    private final static Double VALID_VALUE_MEASUREMENT = 1.0;
-
-    @Test
-    public void testSanitizeNullProperties() throws Exception {
-        Sanitizer.sanitizeProperties(null);
-    }
-
-    @Test
-    public void testSanitizeEmptyProperties() throws Exception {
-        ConcurrentMap<String, String> properties = new ConcurrentHashMap<String, String>();
-        Sanitizer.sanitizeProperties(properties);
-
-        assertTrue(properties.isEmpty());
-    }
-
-    @Test
-    public void testSanitizeValidProperties() throws Exception {
-        ConcurrentMap<String, String> properties = new ConcurrentHashMap<String, String>();
-        properties.put(VALID_KEY_PROPERTY, VALID_VALUE_PROPERTY);
-        Sanitizer.sanitizeProperties(properties);
-
-        assertEquals(properties.size(), 1);
-        assertEquals(properties.get(VALID_KEY_PROPERTY), VALID_VALUE_PROPERTY);
-    }
-
-    @Test
-    public void testSanitizeTooLongKeyAndValueProperties() throws Exception {
-        ConcurrentMap<String, String> properties = new ConcurrentHashMap<String, String>();
-
-        properties.put(createString(Sanitizer.MAX_MAP_NAME_LENGTH + 1), createString(Sanitizer.MAX_VALUE_LENGTH + 1));
-        Sanitizer.sanitizeProperties(properties);
-
-        assertEquals(properties.size(), 1);
-        Map.Entry<String, String> entry = properties.entrySet().iterator().next();
-        assertEquals(entry.getKey().length(), Sanitizer.MAX_MAP_NAME_LENGTH);
-        assertEquals(entry.getValue().length(), Sanitizer.MAX_VALUE_LENGTH);
-    }
-
-    @Test
-    public void testSanitizeIllegalCharsProperties() throws Exception {
-        ConcurrentMap<String, String> properties = new ConcurrentHashMap<String, String>();
-
-        String validKey = createString(10);
-        properties.put("@" + validKey + " ", VALID_VALUE_PROPERTY);
-        Sanitizer.sanitizeProperties(properties);
-
-        assertEquals(properties.size(), 1);
-        Map.Entry<String, String> entry = properties.entrySet().iterator().next();
-        assertEquals(entry.getKey(), validKey);
-        assertEquals(entry.getValue(), VALID_VALUE_PROPERTY);
-    }
-
-    @Test
-    public void testNonValidEmptyUUID() {
-        boolean valid = Sanitizer.isUUID("");
-        assertFalse(valid);
-    }
-
-    @Test
-    public void testNonValidNullUUID() {
-        boolean valid = Sanitizer.isUUID(null);
-        assertFalse(valid);
-    }
-
-    @Test
-    public void testNonValidBadFormat1UUID() {
-        boolean valid = Sanitizer.isUUID("sadfsa");
-        assertFalse(valid);
-    }
-
-    @Test
-    public void testNonValidBadFormat2UUID() {
-        boolean valid = Sanitizer.isUUID("c934153105ac4d8c972e36e97601d5ffc934153105ac4d8c972e36e97601d5ff");
-        assertFalse(valid);
-    }
-
-    @Test
-    public void testValidUUIDWithComma() {
```

Comment:
Is this and all other tests are now covered? I see several things being addressed in the new tests like JSON special characters and length, but not sure if all of those test cases are still addressed in the rest of the tests. If we have code that relies on any of the validated checks, we may want to keep similar tests around.

Reply:
@Dmitry-Matveev I tried to cover all the scenarios which were addressed by the new PR. The rest of the commented tests in this class are related to UUID which we no longer use (Atleast that is what I know).

Reply:
We removed non only UUID tests here but also tests that validate Empty case, Null case, All Valid case and so on and it looks like we replaced them with the single validation of the "JSON needs sanitization" case. We might need to bring some of these tests back (not physically but logically).

Reply:
so for the non empty cases, we have introduced different writer methods write () and writerReq(). Now if writeReq() is called without the required field we populate it with default ones.  I will think if I can add test to check default scenario. Lets discuss on Monday what all other tests can be added in this.